src/core/api/index.js

```diff
@@ -20,7 +20,15 @@ export const category = new Entity('categories', {}, { idAttribute: 'slug' });
 export const user = new Entity('users', {}, { idAttribute: 'username' });
 
 export function makeQueryString(query) {
-  return url.format({ query });
+  const resolvedQuery = { ...query };
+  Object.keys(resolvedQuery).forEach((key) => {
+    if (resolvedQuery[key] === undefined) {
```

Comment:
If sending empty keys triggers bugs (and is also just ugly, I'm all for removing them) we could be more defensive about empty values (eg: `null` and `''` should also cause key deletion).